### PR TITLE
docs: Add kapa.ai to docs

### PR DESCRIPTION
**Pull request by kay-kim**
Add kapa.ai widget to docs. Supersedes https://github.com/MaterializeInc/materialize/pull/26747.

### CODE REVIEW

File: doc/user/layouts/partials/head.html [hunk 1]
```diff
@@ -84,6 +84,22 @@
       container: "#docsearch",
     });
   });
+</script> -->
```
Comment (benesch) [inline doc/user/layouts/partials/head.html:87]: Please just delete this instead of commenting it out—it's in the `git log` if we need it!

**Review by benesch — APPROVED** [review]

File: doc/user/layouts/partials/head.html [hunk 2]
```diff
@@ -81,7 +81,7 @@
   data-modal-title="Materialize"
   data-modal-disclaimer="This is a custom LLM for Materialize with access to developer docs, guides and blog posts. This AI bot is experimental and might provide inaccurate or incomplete answers. Always consult the official Materialize documentation to verify any answers provided here. You can also switch to the Search mode to perform a search instead.
```
Comment (JLDLaughlin) [inline doc/user/layouts/partials/head.html:82]: Is there any way to squish the disclaimer down by a sentence or two? (Sorry I know this is complicated!)

Comment (kay-kim) [reply doc/user/layouts/partials/head.html:82]: Let me try.

Comment (kay-kim) [reply doc/user/layouts/partials/head.html:82]: So ... I think it's a no go.  Adding extra line breaks seems to have zero effect. 😞

Comment (kay-kim) [reply doc/user/layouts/partials/head.html]: but removed the first sentence in the patch.

**Pull request merged**